src/codecs/encoding/encoder.rs

```diff
@@ -93,12 +93,14 @@ impl Encoder<Framer> {
     }
 
     /// Get the suffix that encloses a batch of events.
-    pub const fn batch_suffix(&self) -> &[u8] {
-        match (&self.framer, &self.serializer) {
+    pub const fn batch_suffix(&self, empty: bool) -> &[u8] {
```

Comment:
I think ideally the encoding, for newline delimited framing, would append the newline while encoding each element rather than adding it when finalizing the batch, but that was a bit more refactoring than I was willing to take on at the moment.